src/schema/uuid/thread/resolvers.ts

```diff
@@ -80,15 +81,85 @@ export const resolvers: ThreadResolvers = {
     },
   },
   Mutation: {
+    thread: createMutationNamespace(),
+  },
+  ThreadMutation: {
     async createThread(_parent, payload, { dataSources, userId }) {
       assertUserIsAuthenticated(userId)
       const commentPayload = await dataSources.model.serlo.createThread({
-        ...payload,
+        ...payload.input,
+        userId,
+      })
+      const success = commentPayload !== null
+      return {
+        record:
+          commentPayload !== null
+            ? { __typename: ThreadDataType, commentPayloads: [commentPayload] }
+            : null,
+        success,
+        query: {},
+      }
+    },
+    //TODO: add Caros code
+    async createComment(_parent, payload, { dataSources, userId }) {
+      assertUserIsAuthenticated(userId)
+      const commentPayload = await dataSources.model.serlo.createThread({
+        objectId: 0,
+        ...payload.input,
+        userId,
+      })
+      console.log(commentPayload)
+      return {
+        record: null,
+        success: true,
+        query: {},
+      }
+    },
+    //TODO: make issue to define endpoint needs
+    async setThreadArchived(_parent, payload, { dataSources, userId }) {
+      assertUserIsAuthenticated(userId)
+      const { id, archived } = payload.input
+      const res = await dataSources.model.serlo.archiveThread({
+        id: id,
+        userId,
+        archived,
+      })
+      return {
+        success: res !== null,
+        query: {},
+      }
+    },
+    async setThreadState(_parent, payload, { dataSources, userId }) {
+      assertUserIsAuthenticated(userId)
+      const { id, trashed } = payload.input
+      const firstCommentId = decodeThreadId(id)
+      if (firstCommentId === null)
+        return {
+          success: false,
+          query: {},
+        }
+      const res = await dataSources.model.serlo.setUuidState({
+        id: [firstCommentId],
```

Comment:
```suggestion
        ids: [firstCommentId],
```

Since it is a list, the parameter name should be changed